src/main/java/org/jenkinsci/plugins/mesos/MesosCloud.java

```diff
@@ -263,10 +265,40 @@ public FormValidation doCheckJenkinsUrl(@QueryParameter String jenkinsUrl) {
       }
     }
 
-    /** Test connection from configuration page. */
+    /**
+     * Test connection from configuration page.
+     *
+     * @param mesosMasterUrl The Mesos master URL set by the user.
+     * @return Whether the URL is correct and reachable or a validation error.
+     */
     public FormValidation doTestConnection(
         @QueryParameter("mesosMasterUrl") String mesosMasterUrl) {
-      throw new NotImplementedException("Connection testing is not supported yet.");
+      FormValidation urlValidation = doCheckMesosMasterUrl(mesosMasterUrl);
```

Comment:
does this have to be manually called?

https://github.com/mesosphere/jenkins-mesos-plugin/blob/master/src/main/java/org/jenkinsci/plugins/mesos/MesosCloud.java#L188

would this validate the Mesos master url automatically when the  test connection button is pressed or only if the form is submitted?

Reply:
I don't think it is called automatically. Let me try.

Reply:
..I checked. The validation is not called. However, the try catch will catch the `java.net.MalformedURLException`. I'd keep the validation.